### ISSUE TITLE
Issues with Easy Install

### ISSUE THREAD

**Issue by alejandroiam** [frappe/bench#1435]
Issue: Easy Install

**Do the checklist before filing an issue:**
 - [x] Did you retain the logfile (path of logfile is shared while the script is run)? We definitely **need** the logfile to debug any easy install related issues.
 - [ ] Is this something you can debug and fix? Send a pull request! Bug fixes and documentation fixes are welcome

**Distro Information (Required)**

1. Linux epr 5.10.0-21-amd64 # 1 SMP Debian 5.10.162-1 (2023-01-21) x86_64 GNU/Linux
2. PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"

**Command Run (Required)**
` python3 easy-install.py --prod --email [EMAILPLACEHOLDER]` and `sudo python3 easy-install.py --prod --email [EMAILPLACEHOLDER]`

**Log File (Required)**
```
2023-02-21 14:16:46,786 - INFO - Running Production Setup
2023-02-21 14:16:47,339 - ERROR - Docker Compose generation failed
Traceback (most recent call last):
  File "/home/USER/easy-install.py", line 146, in setup_prod
    subprocess.run(
  File "/usr/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/bin/docker', 'compose', '--project-name', 'frappe', '-f', 'compose.yaml', '-f', 'overrides/compose.mariadb.yaml', '-f', 'overrides/compose.redis.yaml', '-f', 'overrides/compose.https.yaml', '--env-file', '.env', 'config']' returned non-zero exit status 15.

```
**Screenshots**
![image](https://user-images.githubusercontent.com/62250256/220440339-178a31ca-5632-447c-9749-1fdac0ff0701.png)

**Additional context**
N?A

**Possible Solution**
N?A

**Comment by sibidharan** [comment]
Same issue! Very inconsistent.

**Comment by k1meister** [comment]
I have the same, unfortunately.

**Comment by azwalzaman** [comment]
I'm encountering this issue as well running on Ubuntu 22.04 and Python 3.10

**Comment by athul** [comment]
Please refer #1424

**Comment by Kartoffelbreili** [comment]
Has anyone managed to install fresh with the script? I still get the 404 error when trying to visit http://<myserverip> on
`python3 easy-install.py --prod --email myemail.com`
![Screenshot 2023-02-27 163159](https://user-images.githubusercontent.com/120725346/221607310-da276d61-8147-47e0-8cda-2d17f6da04f0.jpg)
I did try both scripts in #1424. It does show up when i open the site on the machine it is installed on, but not when i try to access from other machine.

**Comment by Sanat-Teli-V2STech** [comment]
> Has anyone managed to install fresh with the script? I still get the 404 error when trying to visit http:// on `python3 easy-install.py --prod --email myemail.com` ![Screenshot 2023-02-27 163159](https://user-images.githubusercontent.com/120725346/221607310-da276d61-8147-47e0-8cda-2d17f6da04f0.jpg) I did try both scripts in #1424. It does show up when i open the site on the machine it is installed on, but not when i try to access from other machine.

I have the same issue. I have installed it on actual server I can't access it over internet but if I run curl on the server using internal IP it shows it's running

**Comment by azwalzaman** [comment]
> > Has anyone managed to install fresh with the script? I still get the 404 error when trying to visit http:// on `python3 easy-install.py --prod --email myemail.com` ![Screenshot 2023-02-27 163159](https://user-images.githubusercontent.com/120725346/221607310-da276d61-8147-47e0-8cda-2d17f6da04f0.jpg) I did try both scripts in #1424. It does show up when i open the site on the machine it is installed on, but not when i try to access from other machine.
> 
> I have the same issue. I have installed it on actual server I can't access it over internet but if I run curl on the server using internal IP it shows it's running

I haven't been able to do that either. Try following the single server setup tutorial

**Comment by Sanat-Teli-V2STech** [comment]
@azwalzaman I have used the single server docker setup. It kind of works. It says the website is available on `site1.localhost` so I tried to access it on `site1.<external-IP>` obviously it did not work. So, I tried using just the external IP and that did not work either. Not sure what I am missing honestly.

**Comment by azwalzaman** [comment]
> @azwalzaman I have used the single server docker setup. It kind of works. It says the website is available on `site1.localhost` so I tried to access it on `site1.<external-IP>` obviously it did not work. So, I tried using just the external IP and that did not work either. Not sure what I am missing honestly.

You need to set up some DNS records for that (after getting a domain). And you can't access .localhost from a remote server.

**Comment by Sanat-Teli-V2STech** [comment]
@azwalzaman Can you explain a bit more?
I did add this: `<internal IP>   localhost` to my `/etc/hosts` file

**Comment by azwalzaman** [comment]
> ```
>    @azwalzaman Can you explain a bit more?
> ```
> 
> I did add this: `<internal IP> localhost` to my `/etc/hosts` file

Yeah I never had to do that. After following all the steps in the single server setup guide I got a domain, say `example.com`, and if my ERPNext site was `one.example.com` I would make a subdomain `one.example.com` pointing to my server's IP

**Comment by Sanat-Teli-V2STech** [comment]
okay, I see. Thank you. I will try that.

**Comment by Sanat-Teli-V2STech** [comment]
@azwalzaman I tried the single server docker setup and I am getting `404 page not found.`
![erpnext](https://user-images.githubusercontent.com/122013120/223703405-737aa55c-4166-4dc3-b318-af0389ca64a2.png)

**Comment by azwalzaman** [comment]
> @azwalzaman I tried the single server docker setup and I am getting `404 page not found.` ![erpnext](https://user-images.githubusercontent.com/122013120/223703405-737aa55c-4166-4dc3-b318-af0389ca64a2.png)

What did you name your erpnext site? And what did you name your subdomain? (fake examples) Is traefik working properly?

**Comment by Sanat-Teli-V2STech** [comment]
I have added `TRAEFIK_DOMAIN=frappe.mydomain.com` in the traefik .env file.
and I am getting this from traefik container
```
Cannot issue for \"one.example.com\": The ACME server refuses to issue a certificate for this domain name, because it is forbidden by policy
```

I think I should change this 
```
echo "SITES=\`one.example.com\`,\`two.example.com\`" >> ~/gitops/erpnext-one.env`
 ```
part and replace it with
```
echo "SITES=\`frappe.mydomain.com\`" >> ~/gitops/erpnext-one.env
```

**Comment by azwalzaman** [comment]
> I have added `TRAEFIK_DOMAIN=frappe.mydomain.com` in the traefik .env file. and I am getting this from traefik container
> 
> ```
> Cannot issue for \"one.example.com\": The ACME server refuses to issue a certificate for this domain name, because it is forbidden by policy
> ```
> 
> I think I should change this
> 
> ```
> echo "SITES=\`one.example.com\`,\`two.example.com\`" >> ~/gitops/erpnext-one.env`
> ```
> 
> part and replace it with
> 
> ```
>  `echo "SITES=\`frappe.mydomain.com\`" >> ~/gitops/erpnext-one.env
> ```

Yeah, you're supposed to change some of the values in the guide

**Comment by Kartoffelbreili** [comment]
> Yeah, you're supposed to change some of the values in the guide

If i dont have a domain and i just want to use this in my LAN, what do i need to do starting from the easy-install.py?

**Comment by Sanat-Teli-V2STech** [comment]
> > Yeah, you're supposed to change some of the values in the guide
> 
> If i dont have a domain and i just want to use this in my LAN, what do i need to do starting from the easy-install.py?

If you don't have a domain then I would suggest using the `pwd.yaml` file for testing or instead just don't setup the Traefik part from frappe-docker single server example.

**Comment by OsricDhoubhadel** [comment]
Can confirm the "404 page not found" issue persists on fresh install using easy-install.py . I'm using Ububtu 22 on Google cloud

**Comment by DevOpsFreako** [comment]
@sibidharan @k1meister  @azwalzaman @athul @alejandroiam  @OsricDhoubhadel @sanatv2s @Kartoffelbreili 
Guys If You Found Solution For This Error Then Please Share

**Comment by OsricDhoubhadel** [comment]
> @sibidharan @k1meister @azwalzaman @athul @alejandroiam @OsricDhoubhadel @sanatv2s @Kartoffelbreili Guys If You Found Solution For This Error Then Please Share

Not sure if this is fixed.  I used bench manual installation

**Comment by azwalzaman** [comment]
I think Revant fixed it already?

**Comment by DevOpsFreako** [comment]
> 

Can You Explain What Revant  ?  and if possible can you share your setup file  and any link if that you have try and it work

**Comment by azwalzaman** [comment]
> > 
> 
> Can You Explain What Revant ? and if possible can you share your setup file and any link if that you have try and it work

I stopped using the script and just did the docker setup myself sorry. Should've been fixed in #1424

**Comment by darkerdead** [comment]
The docker setup is nice and easy honestly. I would go for that as well.
For me, the issue was that I forgot to add the domain like here:
`echo "SITES=\`frappe.mydomain.com\`" >> ~/gitops/erpnext-one.env` and after that the error went away.

**Comment by Sanat-Teli-V2STech** [comment]
The docker setup is nice and easy honestly. I would go for that as well.
For me, the issue was that I forgot to add the domain like I had mentioned previously:
```
echo "SITES=\`frappe.mydomain.com\`" >> ~/gitops/erpnext-one.env 
```
and after that the error went away.

**Comment by DevOpsFreako** [comment]
@sibidharan @k1meister @azwalzaman @athul @alejandroiam @OsricDhoubhadel @sanatv2s @Kartoffelbreili
Thank you For your Reply Guys 

I have try your solution but still Getting Same Error 

I'm Doing This Setup In AWS Cloud Ec2 Instance  Ubuntu 22.04.2 LTS
I have Install Both Docker and Docker compose  

rahul@ip-172-31-9-252:~$ docker -v
Docker version 23.0.6, build ef23cbc
rahul@ip-172-31-9-252:~$ docker compose version
Docker Compose version v2.17.2

According To @revant   He said try His script 
This One 
https://github.com/frappe/bench/pull/1424#issuecomment-1434177414

Like I Try 
![image](https://github.com/frappe/bench/assets/128013315/9da9e541-f411-4085-8e9c-7a9d106039c1)

and container are created  
![image](https://github.com/frappe/bench/assets/128013315/f3bab701-bef8-4e64-b23e-631fa38c1f13)

but when access on browser it's give same error  
![image](https://github.com/frappe/bench/assets/128013315/c24b204b-81e8-4b42-aa64-32d422f50dcb)

and he also Said about self certificate  
how to add  like Create a SSL certificate  like we usually do   okey createing certficate how to add with frappe 

Note : I don't change any thing in env file or nor I have purchase domain

If any have complete Steps Then Please Share  or can provide solution for above

**Comment by azwalzaman** [comment]
> @sibidharan @k1meister @azwalzaman @athul @alejandroiam @OsricDhoubhadel @sanatv2s @Kartoffelbreili Thank you For your Reply Guys
> 
> I have try your solution but still Getting Same Error
> 
> I'm Doing This Setup In AWS Cloud Ec2 Instance Ubuntu 22.04.2 LTS I have Install Both Docker and Docker compose
> 
> rahul@ip-172-31-9-252:~$ docker -v Docker version 23.0.6, build ef23cbc rahul@ip-172-31-9-252:~$ docker compose version Docker Compose version v2.17.2
> 
> According To @revant He said try His script This One [#1424 (comment)](https://github.com/frappe/bench/pull/1424#issuecomment-1434177414)
> 
> Like I Try ![image](https://user-images.githubusercontent.com/128013315/237282046-9da9e541-f411-4085-8e9c-7a9d106039c1.png)
> 
> and container are created ![image](https://user-images.githubusercontent.com/128013315/237282738-f3bab701-bef8-4e64-b23e-631fa38c1f13.png)
> 
> but when access on browser it's give same error ![image](https://user-images.githubusercontent.com/128013315/237282436-c24b204b-81e8-4b42-aa64-32d422f50dcb.png)
> 
> and he also Said about self certificate how to add like Create a SSL certificate like we usually do okey createing certficate how to add with frappe
> 
> Note : I don't change any thing in env file or nor I have purchase domain
> 
> If any have complete Steps Then Please Share or can provide solution for above

You can't access it with an IP address. Also, site you used has .localhost but it's on a remote server. That won't work I think

**Comment by Sanat-Teli-V2STech** [comment]
> You can't access it with an IP address. Also, site you used has .localhost but it's on a remote server. That won't work I think

This is true. Please setup the A record for your domain / subdomain before starting with the installation procedure.

**Comment by azwalzaman** [comment]
He can also just run the script locally if he just wants to test it out

**Comment by Sanat-Teli-V2STech** [comment]
Yeah, I just used the `pwd.yaml` to test it out. 

I have a question related to the ssl. I hope it's okay to ask here because I don't want to create a new issue.
So afaik, the SSL certificate is brought by using caddy from Let's Encrypt, correct?  And as per my knowledge Let's Encrypt certificate expires after 3 months. Will caddy renew the certificate automatically or is there something which I will have to do manually? and how will I have to do that?

**Comment by azwalzaman** [comment]
> Yeah, I just used the `pwd.yaml` to test it out.
> 
> I have a question related to the ssl. I hope it's okay to ask here because I don't want to create a new issue. So afaik, the SSL certificate is brought by using caddy from Let's Encrypt, correct? And as per my knowledge Let's Encrypt certificate expires after 3 months. Will caddy renew the certificate automatically or is there something which I will have to do manually? and how will I have to do that?

Yeah it does. @DevOpsFreako If that previous post has your servers real IP please remove it lol

**Comment by Kartoffelbreili** [comment]
Havnt tested in a bit. but since last update is a bit back it should have the same outcome. I COULD run this completely exposed, but im wondering if i can first run this in LAN for a few years before we actually NEED to use all the online features. Can even have a local cert. This was how that ended up btw shortly after the easy-install revamp.

@DevOpsFreako 
![hmmm](https://github.com/frappe/bench/assets/120725346/4910c5c2-896e-457e-b03f-0dca046511ab)
Ah yes, i can see on YOUR server that you are as far as me. 😅

**Comment by DevOpsFreako** [comment]
> @sibidharan @k1meister @azwalzaman @athul @alejandroiam @OsricDhoubhadel @sanatv2s @Kartoffelbreili 

Thank you For your Reply Guys
> 
> I have try your solution but still Getting Same Error
> 
> I'm Doing This Setup In AWS Cloud Ec2 Instance Ubuntu 22.04.2 LTS I have Install Both Docker and Docker compose
> 
> rahul@ip-172-31-9-252:~$ docker -v Docker version 23.0.6, build ef23cbc rahul@ip-172-31-9-252:~$ docker compose version Docker Compose version v2.17.2
> 
> According To @revant He said try His script This One [#1424 (comment)](https://github.com/frappe/bench/pull/1424#issuecomment-1434177414)
> 
> Like I Try ![image](https://user-images.githubusercontent.com/128013315/237282046-9da9e541-f411-4085-8e9c-7a9d106039c1.png)
> 
> and container are created ![image](https://user-images.githubusercontent.com/128013315/237282738-f3bab701-bef8-4e64-b23e-631fa38c1f13.png)
> 
> but when access on browser it's give same error ![image](https://user-images.githubusercontent.com/128013315/237282436-c24b204b-81e8-4b42-aa64-32d422f50dcb.png)
> 
> and he also Said about self certificate how to add like Create a SSL certificate like we usually do okey createing certficate how to add with frappe
> 
> Note : I don't change any thing in env file or nor I have purchase domain
> 
> If any have complete Steps Then Please Share or can provide solution for above


--------------------------------------------------------------

Okay, guys! Now I have both the certificate and the domain.

So, how should I add them and where? Should I start by running the script first, and then go inside the Nginx container or Traefik? Also, where should I add these certificates and the domain?

Please share your knowledge with me or provide any reference links that can help me.

---------------------------------------------------------------------------

**Comment by Sanat-Teli-V2STech** [comment]
I haven't personally tried the script, so, I am not sure how it works but the general idea to manually add the ssl is something like this:
 
1. Create a folder for your certificate and upload the certificate in that folder
2. nginx or apache or whatever web server you're using, stop the webserver
3. add the certificate path to the webserver config file
4. start the web server.

**Comment by DevOpsFreako** [comment]
@sanatv2s 
Create a self-signed certificate.
Stop the Nginx container.
Add the certificate to the Nginx configuration file.
Start the Nginx container.

but stil getting same issue can you try this script 
Where should I add the domain name? Should it be in the /etc/hosts file, the Nginx configuration file, or the Traefik container configuration file (site file)?
Can you share your setup file? I would like to see how you set it up, even if it was done manually in the cloud, not locally.

**Comment by Sanat-Teli-V2STech** [comment]
What is the command you're using to run the script? I was about to try the script but then I saw all the usage flags. Are you using proper usage flags? like `--prod`, `--email`, `--sitename` ? The script automatically adds ssl as well

**Comment by DevOpsFreako** [comment]
@sanatv2s  yes command were right 
python3 easy-install.py --prod --email good@gmail.com
yes it will create default ssl and add but @revant  he said try create self certificate   and i don't create site as deafult site created  site1. localhost

i have try this one 
https://github.com/frappe/bench/pull/1424#issuecomment-1434177414

wget https://raw.githubusercontent.com/revant/bench/fix-easy-install-tls/easy-install.py
python3 easy-install.py --prod --email good@gmail.com

**Comment by Sanat-Teli-V2STech** [comment]
@DevOpsFreako Hey, I think I have figured out the issue thanks to @yashodhank 

You have to add domain name and ip address to `/etc/hosts` like @yashodhank has mentioned [here](https://github.com/frappe/bench/issues/1431#issuecomment-1547024396)

And use the `--site` flag as well for creating the letsencrypt certificate.

Make sure you have the domain and A record for that domain setup beforehand.

**Comment by Kartoffelbreili** [comment]
> Make sure you have the domain and A record for that domain setup beforehand.

I did try it with this dyndns: something.ddnss.ch  I also portforwarded 80 and 443 to my local machine and set up erpnext with:
python3 easy-install.py --prod -s something.ddnss.ch --email Mymail
Good news i am live and have a site exposed. Bad news its the standard traefik cert. What do i do about that?
